checkov/azure_pipelines/runner.py

```diff
@@ -30,3 +34,46 @@ def _parse_file(
 
     def is_workflow_file(self, file_path: str) -> bool:
         return file_path.endswith(('azure-pipelines.yml', 'azure-pipelines.yaml'))
+
+    def get_resource(self, file_path: str, key: str, supported_entities: Iterable[str],
+                     definitions: dict[str, Any] | None = None, root_folder: str | Path | None = None) -> str:
+        relative_file_path = f"/{os.path.relpath(file_path, root_folder)}"
+        if not self.definitions or not isinstance(self.definitions, dict):
+            return relative_file_path
+        start_line, end_line = self.get_start_and_end_lines(key)
+        resource_name = generate_resource_key_recursive(start_line, end_line, self.definitions[file_path])
+        if not resource_name:
+            return relative_file_path
+        return f"{relative_file_path}:{resource_name}"
+
+
+def generate_resource_key_recursive(start_line: int, end_line: int,
+                                    conf: Dict[str, Any] | List[Dict[str, Any]], key: str | None = None
+                                    ) -> str | None:
+    if not isinstance(conf, dict):
+        return key
+
+    def _get_resource_from_dict(dict_to_inspect: dict[str, Any], key: str | None) -> str | None:
+        if dict_to_inspect[START_LINE] <= start_line <= end_line <= dict_to_inspect[END_LINE]:
+            job_name = dict_to_inspect.get('job', None)
+            key = f'{key}.{job_name}' if job_name else key
+            if dict_to_inspect[START_LINE] == start_line:
+                return key
+            return generate_resource_key_recursive(start_line, end_line, dict_to_inspect, key=key)
+        return None
+
+    for k, value in conf.items():
+        if isinstance(value, dict):
+            new_key = f'{key}.{k}' if key else k
+            resource = _get_resource_from_dict(value, new_key)
+            if resource:
+                return resource
+        elif isinstance(value, list):
+            for i, e in enumerate(value):
+                if isinstance(e, dict):
+                    name = f'{k}[{i}]' if k != 'jobs' else k
```

Comment:
it would be great to align the names of steps to the one in GHA
`jobs.container-test-job.steps.1[Check for dockerenv file]`
to get the actual human readable name you would need to check, if there is a key `displayName` with some value

Reply:
Nice didn't know about the `displayName` option, adding it